tests/SpecTests/TransactionsSpecTest.php

```diff
@@ -202,6 +208,84 @@ public function provideTests()
         return $testArgs;
     }
 
+    /**
+     * Prose test 1: Test that starting a new transaction on a pinned
+     * ClientSession unpins the session and normal server selection is performed
+     * for the next operation.
+     */
+    public function testStartingNewTransactionOnPinnedSessionUnpinsSession()
```

Comment:
FWIW, I opened [SPEC-1433](https://jira.mongodb.org/browse/SPEC-1433) to determine if these prose tests are still relevant. I see no harm in adding them, though.